lib/util.js

```diff
@@ -230,17 +230,23 @@ function format(f) {
   return str;
 }
 
-var debugs = {};
-var debugEnviron;
+const debugs = {};
+
+let debugEnvRegex;
+const regExpSpecialChars = /[|\\{}()[\]^$+?.]/g;
+if (debugEnvRegex === undefined) {
```

Comment:
This `if` statement will always evaluates to true. It should actually test for `process.env.NODE_DEBUG` and only if that is truthy it should be evaluated.

This is how I would write it
```js
const debugs = {};
let debugEnvRegex = /^$/;
if (process.env.NODE_DEBUG) {
  const debugEnv = process.env.NODE_DEBUG
    // Escape RegExp special characters
    .replace(/[|\\{}()[\]^$+?.]/g, '\\$&')
    .replace(/\*/g, '.*')
    .replace(/,/g, '$|^')
    .toUpperCase();
  debugEnvRegex = new RegExp(`^${debugEnv}$`, 'i');
}
```

Reply:
You are right! I should consider that corner case at the first time. Thx : )